module/profiler/profiler.go

```diff
@@ -143,11 +146,96 @@ func newProfileFunc(name string) profileFunc {
 	}
 }
 
-func (p *AutoProfiler) pprofAllocs(w io.Writer) error {
+func (p *AutoProfiler) goHeapProfile() (*profile.Profile, error) {
```

Comment:
pass in the wanted type list instead of using the hardcode list below.  looks like allocs don't care about "inuse_objects" and "inuse_space"